### PR TITLE
feat(pack): add hyprland config file language support

### PR DESCRIPTION
**Pull request by WilliamBy**

### CODE REVIEW

**Comment by mehalter** [comment]
We should hold off until this PR is merged: https://github.com/williamboman/mason-lspconfig.nvim/pull/428

File: lua/astrocommunity/pack/hyprland/init.lua
Comment (mehalter) [inline lua/astrocommunity/pack/hyprland/init.lua:51]:
```suggestion
```

This should be handled by `nvim-lspconfig`

File: lua/astrocommunity/pack/hyprland/init.lua
Comment (mehalter) [inline lua/astrocommunity/pack/hyprland/init.lua:54]:
```suggestion
  {
    "AstroNvim/astrolsp",
    opts = function(_, opts)
      local astrocore = require "astrocore"
      opts.servers = astrocore.list_insert_unique(opts.servers, { "hyprls" })

      return astrocore.extend_tbl(opts, {
        config = {
          hyprls = { filetypes = { "hyprlang" } },
        },
      })
    end,
  },
```

This tells AstroLSP to configure the server in `lspconfig` since it's not currently supported in `mason-lspconfig` as well as configure the filetypes correctly. Once that is merged in, we can simplify this to:

```lua
  {
    "AstroNvim/astrolsp",
    opts = {
      config = {
        hyprls = { filetypes = { "hyprlang" } },
      },
    },
  },
```

**Review by mehalter — CHANGES_REQUESTED** [review]
A few fixes/improvements

**Comment by mehalter** [comment]
Once the PR to `mason-lspconfig` is merged we should simplify the AstroLSP configuration as well as add:

```lua
  {
    "williamboman/mason-lspconfig.nvim",
    optional = true,
    opts = function(_, opts)
      opts.ensure_installed = require("astrocore").list_insert_unique(opts.ensure_installed, { "hyprls" })
    end,
  },
```

File: lua/astrocommunity/pack/hyprland/init.lua
Comment (mehalter) [inline lua/astrocommunity/pack/hyprland/init.lua:27]:
```suggestion
          [".*/hypr/.*.conf"] = "hyprlang",
          ["hypr.*.conf"] = "hyprlang",
```

These patterns do not work, they are real regular expressions so this fixes them.

File: lua/astrocommunity/pack/hyprland/README.md
Comment (WilliamBy) [inline lua/astrocommunity/pack/hyprland/README.md:8]:
```suggestion
    - `*/hypr/*.conf`, `hypr.*.conf`
```
this fix wrong pattern form

File: lua/astrocommunity/pack/hyprland/init.lua
Comment (WilliamBy) [inline lua/astrocommunity/pack/hyprland/init.lua]:
```suggestion
          [".*/hypr/.*.conf"] = "hyprlang",
          ["hypr.*.conf"] = "hyprlang",
```

Comment (mehalter) [reply lua/astrocommunity/pack/hyprland/init.lua]: yeah they have

File: lua/astrocommunity/pack/hyprland/README.md
Comment (mehalter) [inline lua/astrocommunity/pack/hyprland/README.md:9]:
```suggestion
```

also this

**Review by mehalter — APPROVED** [review]

**Comment by mehalter** [comment]
Ok, I just did a follow up revision of this PR to rename this to `hyprlang` pack just because that is technically the name of the configuration language rather than `hyprland`

**Comment by mehalter** [comment]
Thanks for the contribution! This is a great one! :D Love seeing packs getting added that really decrease setup like setting up filetypes and things 😄

**Pull request merged**
